library/panic_unwind/src/seh.rs

```diff
@@ -220,7 +220,7 @@ extern "C" {
 // This is fine since the MSVC runtime uses string comparison on the type name
 // to match TypeDescriptors rather than pointer equality.
 static mut TYPE_DESCRIPTOR: _TypeDescriptor = _TypeDescriptor {
-    pVFTable: unsafe { &TYPE_INFO_VTABLE } as *const _ as *const _,
```

Comment:
Yes, it said something about external statics iirc

Reply:
> Strange, did this one not trigger the warning? It doesn't seem to be inside any of the `allow`.

Ah, I guess this is because it is not a `mut` static:
https://github.com/rust-lang/rust/pull/121303/files#diff-1308e363d41cbb1f3f93d560e000263610240cc456596918347038d303dac7b6L214

Reply:
Ah I see, you just made them all use addr_of for good measure. Makes sense!

Reply:
> Ah I see, you just made them all use addr_of for good measure. Makes sense!

I think there are many more places where `& ... as *const ...` is used. Would it make sense to implement a lint (probably allow-by-default) suggesting to replace those with `addr_of!`? Or maybe at least for me to go through the libs and change those manually?

Reply:
Yeah those should all be addr_of ideally.